src/components/table/Selector.tsx

```diff
@@ -0,0 +1,36 @@
+import { h } from "@stencil/core"
+
+export class Selector<T> {
+	#selected?: T[]
+	get selected(): T[] {
+		return (this.#selected ??= Object.values(this.data))
+	}
+	private constructor(private readonly key: keyof T, private data: Readonly<Record<string, T>> = {}) {}
+	handle(data: Record<string, T>): Selector<T> {
+		return new Selector(
+			this.key,
+			"check-all" in data
+				? data["check-all"] ?? {}
+				: Object.fromEntries(Object.entries({ ...this.data, ...data }).filter(([_, value]) => value != undefined))
+		)
+	}
+	render(item: T | T[]): any {
```

Comment:
this funktion returns JSX. The proper return type for this would be `HTMLElement` which would be better to use than `any`. If you want to be more precise you can probably use `HTMLSmoothlyCheckboxElement`